test/jdk/sun/net/www/protocol/https/HttpsURLConnection/CloseKeepAliveCached.java

```diff
@@ -208,25 +207,66 @@ public boolean verify(String hostname, SSLSession session) {
     volatile Exception clientException = null;
 
     public static void main(String args[]) throws Exception {
+        separateServerThread = Boolean.parseBoolean(args[0]);
+        System.out.printf("separateServerThread: %s%n", separateServerThread);
+
         String keyFilename =
-            System.getProperty("test.src", "./") + "/" + pathToStores +
-                "/" + keyStoreFile;
+                System.getProperty("test.src", "./") + "/" + pathToStores +
+                        "/" + keyStoreFile;
         String trustFilename =
-            System.getProperty("test.src", "./") + "/" + pathToStores +
-                "/" + trustStoreFile;
+                System.getProperty("test.src", "./") + "/" + pathToStores +
+                        "/" + trustStoreFile;
 
         System.setProperty("javax.net.ssl.keyStore", keyFilename);
         System.setProperty("javax.net.ssl.keyStorePassword", passwd);
         System.setProperty("javax.net.ssl.trustStore", trustFilename);
         System.setProperty("javax.net.ssl.trustStorePassword", passwd);
 
-        if (debug)
-            System.setProperty("javax.net.debug", "all");
+        System.setProperty("javax.net.debug", "all");
+
+        // setting up the error stream for further analysis
+        var errorCapture = new ByteArrayOutputStream();
+        var errorStream = new PrintStream(errorCapture);
+        var originalErr = System.err; // saving the initial error stream, so it can be restored
+        System.setErr(errorStream);
 
         /*
          * Start the tests.
          */
-        new CloseKeepAliveCached();
+        try {
+            new CloseKeepAliveCached();
+        } finally {
+            // this will allow the error stream to be printed in case of an exception inside for debugging purposes
+            System.setErr(originalErr);
+            if (debug) {
```

Comment:
This field is hard-coded to false which requires rebuilding to turn it on. I have been changing that to:

`static boolean debug = Boolean.getBoolean("test.debug");`

So it can be turned on with a system property.